activerecord/test/cases/test_case.rb

```diff
@@ -125,7 +125,7 @@ def with_db_warnings_action(action, warnings_to_ignore = [])
 
       yield
     ensure
-      ActiveRecord.db_warnings_action = @original_db_warnings_action
+      ActiveRecord.db_warnings_action = @original_db_warnings_action || :ignore
```

Comment:
This actually might be another small bug to fix.

`ActiveRecord.db_warnings_action` is a reader, but the writer is a method defined here:
https://github.com/rails/rails/blob/757f4896417f86b58303e416f4040a2b434abdd6/activerecord/lib/active_record.rb#L216-L217

The default value is `:ignore` which the case sets to `nil`. So the reader returns `nil`, but that is not a valid value to set. So here when we read it and then reassigned it actually hits the `else` case and raises. This is probably just a bug in test, but the solution could also be to set the `:ignore` case to:

```ruby
      when :ignore, nil
        nil
```

Reply:
Why wasn't those tests raising an error then? I don't think we should should accept `nil` in the writer, but we should have seen an exception here.

Reply:
Is this a bug? `@original_db_warnings_action` is set to `:ignore` in the setup block for our tests, it's not actually returning the value read from the attribute, so it should be okay as-is.

Reply:
Yeah I think you're right. CI caught it that `@original_db_warnings_action` was set in one file but read in another. I'll tidy this up. Thanks.